### ISSUE TITLE
100% height intermittently ignored

### ISSUE THREAD

**Issue by mattspeller** [whawker/react-jsx-highcharts#61]
Hello

I have followed the Highchart documentation to set the TreeMap larger than the default 400px, but setting the height in the chart element to 100%.

However I see that no effect for this setting (immediately at least). Sometimes the grid re-draws correctly to 100% but most times it stays at the default 400px tall. 

Please see below for an example, I would not expect to see any of the red region if the grid expand as expected. 

https://codesandbox.io/s/5vrw6rv9ln

Any thoughts?

Matt

**Comment by whawker** [comment]
I'm not sure this is a React JSX Highcharts issue, the same appears to happen when [using the Highcharts library directly](http://jsfiddle.net/t6mfkkn9/)?

**Comment by mattspeller** [comment]
hmmm.. not sure about that. Here is a link to a fiddle I copied from the highcharts demo samples with the height set to 100% - you can see it correctly sizes immediately.

http://jsfiddle.net/xz76723g/

It's definitely a limitation of the react wrapper around a charting component. I have seen the same for a column series and a treemap.

**Comment by whawker** [comment]
It's likely due to how the chart options are applied _after_ the chart is initialized.

As a work around, you could ensure they are set at initialization, by passing them on the `HighchartsChart` object

```jsx
const initChartOptions = { height: '100%' };

return (
  <HighchartsChart chart={initChartOptions}>
    //...
  </HighchartsChart>
);
```

**Comment by mattspeller** [comment]
thanks Will - perfect - that worked on the example I gave above. 

Of course, real life cases are slightly more complex than a sandbox example so I thought I would put down here what I found out to help you and others if sizing comes up again.

- Highcharts Chart object can be sized either with number or % and it works differently based on choice.
  - If you set the height to be a constant number, it will set the chart to be that height.
  - If you set the height to be a %, it will set the chart to be that % ratio of the offset width of the container it held within.
  - Do not set height to be '100px' as it will be ignored, it is either {100} or '100%' for example.

[documentation on width](https://api.highcharts.com/highcharts/chart.width)
[documentation on height](https://api.highcharts.com/highcharts/chart.height)

So this means that by setting the height to be 100% may very well not set the height if the container it is held within is does not have a height set. In my case, I am using [Golden Layout](http://golden-layout.com/) with the React Component hosted within each dockable panel. Each container (in react terms) has a height of NaN and a width of NaN when inspecting the DOM.

If you render a Highcharts control within a container without a height and width set but initialise the chart options to 100%, you get exceptions in the console like this
`attribute height: Expected length, "NaN".`

So, to make the height configurable and fill the panel within GoldenLayout dockable containers, I needed to pass down the dockable container width and height to the Chart and set the size accordingly  to the specific number like this:

Container:
`
mapStateToProps(state, props){
...
    height: props.glContainer.height,
    width: props.glContainer.width,
...
}
`

React Component
`
render(){
    const initChartOptions = { height: this.props.height, width: this.props.width };
    return (
       ...
        <HighchartsChart callback={this.getChart} chart={initChartOptions}>
          <Chart height={this.props.height} width={this.props.width} />
       ...
    );
}
`

interestingly, I needed to set the height in both the chart options and also set the height on the chart object to get this to consistently work. I guessing, like you said, orders things applied varies based on speed of render.

Hope this helps someone!
